protocol/x/affiliates/keeper/keeper.go

```diff
@@ -162,41 +179,43 @@ func (k Keeper) GetTierForAffiliate(
 	numTiers := uint32(len(affiliateTiers.GetTiers()))
 	maxTierLevel := numTiers - 1
 	currentTier := uint32(0)
-	referredVolume, exists := k.GetReferredVolume(ctx, affiliateAddr)
-
-	if !exists {
-		// If referred volume is not found, set it to 0.
-		referredVolume = dtypes.NewInt(0)
+	referredVolume, err := k.GetReferredVolume(ctx, affiliateAddr)
+	if err != nil {
+		return 0, 0, err
 	}
 
-	for _, tier := range affiliateTiers.GetTiers() {
-		if referredVolume.BigInt().Int64() >= int64(tier.ReqReferredVolume) {
-			currentTier = tier.GetLevel()
+	for index, tier := range affiliateTiers.GetTiers() {
+		if referredVolume.Int64() >= int64(tier.ReqReferredVolume) {
+			// safe to do as tier cannot be negative
+			currentTier = uint32(index)
 		}
 	}
 
 	if currentTier == maxTierLevel {
 		return currentTier, affiliateTiers.GetTiers()[currentTier].TakerFeeSharePpm, nil
 	}
+
 	numCoinsStaked := k.statsKeeper.GetStakedAmount(ctx, affiliateAddr)
-	for _, tier := range affiliateTiers.GetTiers() {
-		if numCoinsStaked.Cmp(big.NewInt(
-			int64(affiliateTiers.GetTiers()[currentTier].ReqStakedWholeCoins))) >= 0 &&
-			tier.GetLevel() > currentTier {
-			currentTier = tier.GetLevel()
+	for i := currentTier + 1; i < numTiers; i++ {
+		expMultiplier := new(big.Int).Exp(
+			big.NewInt(10),
+			big.NewInt(-lib.BaseDenomExponent),
+			nil,
+		)
+		reqStakedCoins := new(big.Int).Mul(
+			big.NewInt(int64(affiliateTiers.GetTiers()[i].ReqStakedWholeCoins)),
+			expMultiplier,
+		)
+		if numCoinsStaked.Cmp(reqStakedCoins) >= 0 {
```

Comment:
The `AffiliateTiers` object should hold the invariant that `ReqStakedWholeCoins` are strictly increasing (could you add a TODO item to check this invariant in the message handler for `MsgUpdateAffiliateTiers`), so as soon as as you see `numCoinsStaked.Cmp(reqStakedCoins) < 0` we can just break.

Reply:
Just to double check, do we have a TODO item for the variant?